Orange/widgets/data/owfile.py

```diff
@@ -291,8 +325,19 @@ def browse_file(self, in_demos=0):
             self.recent_paths.remove(recent)
 
         self.recent_paths.insert(0, recent)
-        self.set_file_list()
-        self.open_file(self.recent_paths[0].abspath)
+
+    @staticmethod
+    def is_url_valid(url):
+        try:
+            return bool(Table.from_url(url)), ""
```

Comment:
This test is too expensive if data is large. I think we should just check whether the link works and close the connection without transferring any data.